src/trace-plugin-loader.js

```diff
@@ -116,16 +116,20 @@ function activate(agent) {
       if (!patchSet) {
         // Load the plugin object
         var plugin = originalModuleLoad(instrumentation.file, module, false);
-        patchSet = [];
+        patchSet = {};
```

Comment:
We cannot assume a single version unfortunately. Things like `redis` that are used for caching may be required at different versions for by different dependencies of the primary project.